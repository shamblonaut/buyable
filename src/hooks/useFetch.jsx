import { useEffect, useState } from "react";

const useFetch = (url) => {
  const [data, setData] = useState(null);
  const [error, setError] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    fetch(url)
      .then((response) => {
        if (!response.ok) {
          throw new Error(
            `Error ${response.status}${response.statusText ? `: ${response.statusText}` : ""}`,
          );
        }

        return response.json();
      })
      .then((response) => setData(response))
      .catch((error) => setError(error.message))
      .finally(() => setLoading(false));
  }, [url]);

  return { data, error, loading };
};

export default useFetch;
