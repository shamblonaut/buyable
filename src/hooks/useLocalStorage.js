import { useEffect, useState } from "react";

const useLocalStorage = (name, initialValue = null) => {
  const [value, setValue] = useState(() => {
    const storedValue = localStorage.getItem(name);
    return storedValue !== null ? JSON.parse(storedValue) : initialValue;
  });

  useEffect(() => {
    localStorage.setItem(name, JSON.stringify(value));
  }, [name, value]);

  return [
    value,
    (newValue) =>
      setValue((currentValue) =>
        typeof newValue === "function" ? newValue(currentValue) : newValue,
      ),
  ];
};

export default useLocalStorage;
