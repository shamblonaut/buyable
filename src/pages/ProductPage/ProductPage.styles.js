import styled, { createGlobalStyle, css } from "styled-components";

export const GlobalStyle = createGlobalStyle(css`
  body {
    background-color: white;
  }
`);

export const ImageContainer = styled.div`
  display: flex;
  justify-content: center;
  margin: 16px;

  & img {
    width: auto;
    max-height: 30vh;
  }
`;

export const Content = styled.div`
  flex: 1;
  padding: 16px;
  padding-top: 0;

  background-color: var(--color-background);
  border: 3px solid var(--color-dark);
  border-radius: 16px 16px 0 0;
  border-bottom: none;

  display: flex;
  align-items: center;
  flex-direction: column;
  gap: 16px;
`;

export const Notch = styled.div`
  margin: 6px auto;
  width: 80px;
  height: 12px;
  border: 3px solid var(--color-dark);
  border-radius: 6px;
  background-color: var(--color-light);
`;

export const Details = styled.div`
  flex: 1;
  width: 100%;

  display: flex;
  flex-direction: column;
  gap: 8px;
`;

export const Title = styled.p`
  font-size: 1.75rem;
  font-weight: 700;
`;

export const Price = styled.p`
  font-size: 1.5rem;
  font-weight: 500;
  font-family: var(--font-highlight), sans-serif;
`;

export const Description = styled.p`
  font-size: 1.1rem;
`;

export const CartActionContainer = styled.div`
  display: flex;
  justify-content: center;
  width: max(50%, 360px);
`;
