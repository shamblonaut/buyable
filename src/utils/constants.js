export const AppPosition = Object.freeze({
  HOME: 0,
  SHOP: 1,
  CART: 2,
});
