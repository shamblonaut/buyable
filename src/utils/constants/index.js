import { AppPosition } from "./App";

export { AppPosition };
