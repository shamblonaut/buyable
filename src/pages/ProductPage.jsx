import { useParams } from "react-router-dom";
import styled from "styled-components";

import useFetch from "../hooks/useFetch";

const ImageContainer = styled.div`
  display: flex;
  justify-content: center;
  margin: 16px;

  & img {
    width: auto;
    max-height: 360px;
  }
`;

const Info = styled.p`
  text-align: center;
`;

const ProductPage = () => {
  const { id } = useParams();

  const {
    data: product,
    error,
    loading,
  } = useFetch(`https://fakestoreapi.com/products/${id}`);

  return (
    <main>
      {loading ? (
        <Info>Loading product...</Info>
      ) : error ? (
        <Info>Product not found!</Info>
      ) : (
        product && (
          <>
            <ImageContainer>
              <img src={product.image} alt={product.title} />
            </ImageContainer>
            <h2>{product.title}</h2>
            <p>${product.price}</p>
          </>
        )
      )}
    </main>
  );
};

export default ProductPage;
