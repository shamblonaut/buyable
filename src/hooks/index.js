import useFetch from "./useFetch/useFetch";
import useLocalStorage from "./useLocalStorage/useLocalStorage";
import useCart from "./useCart/useCart";

export { useFetch, useLocalStorage, useCart };
