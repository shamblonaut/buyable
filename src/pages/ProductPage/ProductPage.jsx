import { useOutletContext, useParams } from "react-router-dom";

import { useFetch } from "@/hooks";

import { BackButton, CartActions, Rating } from "@/components";
import { Page, Info } from "@/styles";
import {
  GlobalStyle,
  ImageContainer,
  Content,
  Notch,
  Details,
  Title,
  Price,
  Description,
  CartActionContainer,
} from "./ProductPage.styles";

const ProductPage = () => {
  const { id } = useParams();

  const {
    data: product,
    error,
    loading,
  } = useFetch(`https://fakestoreapi.com/products/${id}`);

  const { cart, setCart, appPosition } = useOutletContext();

  return (
    <Page>
      <BackButton appPosition={appPosition} />
      {loading ? (
        <Info>Loading product...</Info>
      ) : error ? (
        <Info>Product not found!</Info>
      ) : (
        product && (
          <>
            <GlobalStyle />
            <ImageContainer>
              <img src={product.image} alt={product.title} />
            </ImageContainer>
            <Content>
              <Notch />
              <Details>
                <Title>{product.title}</Title>
                <Price>${product.price}</Price>
                <Rating
                  rating={product.rating.rate}
                  count={product.rating.count}
                />
                <Description>{product.description}</Description>
              </Details>
              <CartActionContainer>
                <CartActions product={product} cart={cart} setCart={setCart} />
              </CartActionContainer>
            </Content>
          </>
        )
      )}
    </Page>
  );
};

export default ProductPage;
