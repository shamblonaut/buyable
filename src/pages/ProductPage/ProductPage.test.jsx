import { describe, it, expect, vi, beforeEach } from "vitest";
import { render, screen } from "@testing-library/react";

import { AppPosition } from "@/utils/constants";

import { mockProductList, mockCart } from "@tests/data";
import OutletWrapper from "@tests/components/OutletWrapper";

import ProductPage from "./ProductPage";

const mockOutletContext = {
  cart: mockCart,
  setCart: vi.spyOn(
    {
      setCart: vi.fn(),
    },
    "setCart",
  ),
  appPosition: AppPosition.SHOP,
};
const mockProduct = mockProductList[2];

vi.mock("@/hooks", async (importOriginal) => {
  const actual = await importOriginal();
  return {
    ...actual,
    useFetch: vi.spyOn(
      {
        useFetch: vi.fn(() => ({
          data: mockProduct,
          error: null,
          loading: false,
        })),
      },
      "useFetch",
    ),
  };
});

import { useFetch } from "@/hooks";

describe("Product Page", () => {
  beforeEach(() => {
    vi.clearAllMocks();
  });

  it("renders correctly", () => {
    const { container } = render(
      <OutletWrapper context={mockOutletContext}>
        <ProductPage />
      </OutletWrapper>,
    );

    expect(container).toMatchSnapshot();
  });

  it("links to correct back page", () => {
    render(
      <OutletWrapper context={mockOutletContext}>
        <ProductPage />
      </OutletWrapper>,
    );

    expect(screen.getByRole("link")).toHaveAttribute("href", "/shop");
  });

  it("shows loading text when data is loading", () => {
    useFetch.mockReturnValueOnce({
      data: null,
      error: null,
      loading: true,
    });

    render(
      <OutletWrapper context={mockOutletContext}>
        <ProductPage />
      </OutletWrapper>,
    );

    expect(
      screen.getByText("loading product", { exact: false }),
    ).toBeInTheDocument();
  });

  it("shows error text when there is an error in loading the data", () => {
    useFetch.mockReturnValueOnce({
      data: null,
      error: new Error("Mock Error"),
      loading: false,
    });

    render(
      <OutletWrapper context={mockOutletContext}>
        <ProductPage />
      </OutletWrapper>,
    );

    expect(
      screen.getByText("product not found", { exact: false }),
    ).toBeInTheDocument();
    expect(
      screen.getByText("mock error", { exact: false }),
    ).toBeInTheDocument();
  });

  it("uses correct product for cart actions", () => {
    render(
      <OutletWrapper context={mockOutletContext}>
        <ProductPage />
      </OutletWrapper>,
    );

    expect(screen.getByLabelText("cart-quantity-selector")).toBeInTheDocument();
    expect(
      screen.getByRole("textbox", { name: "custom-quantity-input" }).value,
    ).toBe("2");
  });
});
