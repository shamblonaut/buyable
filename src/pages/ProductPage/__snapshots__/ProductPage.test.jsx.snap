// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Product Page > renders correctly 1`] = `
<div>
  <main
    class="sc-hwkwBN sc-dTWiOz kikxJF hImdCS"
  >
    <a
      class="sc-cpclqO bGNFfF"
      data-discover="true"
      href="/shop"
    >
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-left"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m15 18-6-6 6-6"
        />
      </svg>
    </a>
    <div
      class="sc-eJoXEY hRJqCK"
    >
      <img
        alt="Element S12 8GB RAM 256GB ROM Smartphone"
        src="http://example.com/image-3.jpg"
      />
    </div>
    <div
      class="sc-goiVcJ hLhGRp"
    >
      <div
        class="sc-gSifMm iiajZE"
      />
      <div
        class="sc-ixcdjX hMuNHQ"
      >
        <p
          class="sc-eQaGpr gLtFkG"
        >
          Element S12 8GB RAM 256GB ROM Smartphone
        </p>
        <p
          class="sc-VLMBG jgMLsR"
        >
          $
          249
        </p>
        <div
          class="sc-gahYZc hnMRQX"
        >
          <div
            class="sc-bSFBcf kKclkr"
          >
            <div
              class="sc-gsJsQu kRzqpM"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-star sc-ibashp jUYxjU"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              <div
                class="sc-blIAwI jcoGEn"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-star sc-itBLYH hBkIGd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-gsJsQu kRzqpM"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-star sc-ibashp jUYxjU"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              <div
                class="sc-blIAwI jcoGEn"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-star sc-itBLYH hBkIGd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-gsJsQu kRzqpM"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-star sc-ibashp jUYxjU"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              <div
                class="sc-blIAwI hydHUV"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-star sc-itBLYH hBkIGd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-gsJsQu kRzqpM"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-star sc-ibashp jUYxjU"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              <div
                class="sc-blIAwI faYJSK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-star sc-itBLYH hBkIGd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                  />
                </svg>
              </div>
            </div>
            <div
              class="sc-gsJsQu kRzqpM"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-star sc-ibashp jUYxjU"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                />
              </svg>
              <div
                class="sc-blIAwI faYJSK"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-star sc-itBLYH hBkIGd"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M11.525 2.295a.53.53 0 0 1 .95 0l2.31 4.679a2.123 2.123 0 0 0 1.595 1.16l5.166.756a.53.53 0 0 1 .294.904l-3.736 3.638a2.123 2.123 0 0 0-.611 1.878l.882 5.14a.53.53 0 0 1-.771.56l-4.618-2.428a2.122 2.122 0 0 0-1.973 0L6.396 21.01a.53.53 0 0 1-.77-.56l.881-5.139a2.122 2.122 0 0 0-.611-1.879L2.16 9.795a.53.53 0 0 1 .294-.906l5.165-.755a2.122 2.122 0 0 0 1.597-1.16z"
                  />
                </svg>
              </div>
            </div>
          </div>
          <span
            class="sc-bEjUoa iDEzGJ"
          >
            <p
              class="sc-boKDdR hXXtkX"
            >
              2.3
            </p>
            <p
              class="sc-fOOuSg dWMSsu"
            >
              (
              32
              )
            </p>
          </span>
        </div>
        <p
          class="sc-gXbNzB flnIlw"
        >
          Take stunning pictures, play high-end games without a hitch, and be your best buddy wherever you go. The Element S12 can do it all!
        </p>
      </div>
      <div
        class="sc-dcMTLQ jnAXJp"
      >
        <div
          class="sc-etsjJW eKirvu"
        >
          <div
            aria-label="Edit cart quantity"
            class="sc-kUouGy kcFQSo"
            role="region"
          >
            <button
              aria-label="Decrement cart quantity"
              class="sc-bjMIFn sc-dClGHI gycoPx jTKwYk"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-minus"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
              </svg>
            </button>
            <input
              aria-label="Enter custom cart quantity"
              class="sc-bAehkN kpqawf"
              type="text"
              value="2"
            />
            <button
              aria-label="Increment cart quantity"
              class="sc-bjMIFn sc-fhfEft gycoPx ejGQKl"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-plus"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 12h14"
                />
                <path
                  d="M12 5v14"
                />
              </svg>
            </button>
          </div>
          <button
            aria-label="Remove from cart"
            class="sc-fYmhhH sc-DZJJV bDLkig gLaRVb"
          >
            <svg
              aria-hidden="true"
              class="lucide lucide-trash2 lucide-trash-2"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 6h18"
              />
              <path
                d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
              />
              <path
                d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
              />
              <line
                x1="10"
                x2="10"
                y1="11"
                y2="17"
              />
              <line
                x1="14"
                x2="14"
                y1="11"
                y2="17"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </main>
</div>
`;
